### ISSUE TITLE
Compilation failure on TPU

### ISSUE THREAD

**Issue by fishyboy** [google/automl#488]
During training efficientdet d7 on google colab tpu with my custom dataset, i encountered this problem:

I0607 16:13:53.142851 140561397208832 tpu_estimator.py:540] Starting outfeed thread controller.
INFO:tensorflow:Enqueue next (100) batch(es) of data to infeed.
I0607 16:13:57.123712 140563857274752 tpu_estimator.py:600] Enqueue next (100) batch(es) of data to infeed.
INFO:tensorflow:Dequeue next (100) batch(es) of data from outfeed.
I0607 16:13:57.124172 140563857274752 tpu_estimator.py:604] Dequeue next (100) batch(es) of data from outfeed.
ERROR:tensorflow:Error recorded from outfeed: From /job:worker/replica:0/task:0:
Bad hardware status: Queue is closed.
	 [[node OutfeedDequeueTuple (defined at /usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py:1748) ]]

**ERROR:tensorflow:Error recorded from training_loop: From /job:worker/replica:0/task:0:
Compilation failure: Detected unsupported operations when trying to compile graph _functionalize_body_0[] on XLA_TPU_JIT: Const (No registered 'Const' OpKernel for XLA_TPU_JIT devices compatible with node node box_net/box-4-bn-7/batchnorm/add/y (defined at /usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py:1748) 
	 (OpKernel was found, but attributes didn't match) Requested Attributes: _xla_inferred_shapes=[[]], dtype=DT_HALF, value=Tensor<type: half shape: [] values: 0.00100040436>, _device="/device:TPU_REPLICATED_CORE")node box_net/box-4-bn-7/batchnorm/add/y (defined at /usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py:1748) 
	 [[LoopCond]]
	TPU compilation failed**
	 [[tpu_compile_succeeded_assert/_13283801156586978321/_2629]]
E0607 16:15:01.623807 140563857274752 error_handling.py:75] Error recorded from training_loop: From /job:worker/replica:0/task:0:
Compilation failure: Detected unsupported operations when trying to compile graph _functionalize_body_0[] on XLA_TPU_JIT: Const (No registered 'Const' OpKernel for XLA_TPU_JIT devices compatible with node node box_net/box-4-bn-7/batchnorm/add/y (defined at /usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py:1748) 
	 (OpKernel was found, but attributes didn't match) Requested Attributes: _xla_inferred_shapes=[[]], dtype=DT_HALF, value=Tensor<type: half shape: [] values: 0.00100040436>, _device="/device:TPU_REPLICATED_CORE")node box_net/box-4-bn-7/batchnorm/add/y (defined at /usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/ops.py:1748) 
	 [[LoopCond]]
	TPU compilation failed
	 [[tpu_compile_succeeded_assert/_13283801156586978321/_2629]]
INFO:tensorflow:training_loop marked as finished
I0607 16:15:01.624117 140563857274752 error_handling.py:101] training_loop marked as finished
WARNING:tensorflow:Reraising captured error
W0607 16:15:01.624293 140563857274752 error_handling.py:135] Reraising captured error

the command was:
!python main.py --mode=train \
    --training_file_pattern=gs://pathtotrain/{file_pattern} \
    --eval_name=evaluation \
    --validation_file_pattern=gs://pathtoval/{val_pattern} \
    --val_json_file=tfrecord/json_val.json \
    --model_name={MODEL} \
    --ckpt=gs://effbuck/efficient/{MODEL} \
    --model_dir=gs://*/model_dir/{MODEL}-finetune \
    --train_batch_size=8 \
    --eval_batch_size=8 --eval_samples={val_per_epoch}  \
    --num_examples_per_epoch={images_per_epoch}  --num_epochs=20  \
    --hparams="num_classes=5,moving_average_decay=0,precision=mixed_float16" \
    --strategy="tpu" \
    --eval_master={TPU_ADDRESS} \
    --tpu={TPU_ADDRESS} \

TF version 1.15.1
i have successsfully trained d6 with batchsize of 8 and without mixed_float16
as per d7, i have to use batchsize 8 with mixed_float16 or rather OOM

**Comment by fishyboy** [comment]
By the way, with everything properly set, evaluation of  d6 on colab tpu always said that restoring from ckpt failed due to probably change in model graph, but on colab gpu it just worked well with same config.

**Comment by fsx950223** [comment]
mixed_bfloat16

**Comment by fishyboy** [comment]
and as well if i set main.py --mode=train _and_eval and --eval_master tpu address,  

> mixed_bfloat16

working well, thanks! closing this
